workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/executor/request/client/HttpClient.java

```diff
@@ -0,0 +1,96 @@
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpResponse;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+@Generated
+@SuppressWarnings("unchecked")
+@Component
+public class HttpClient {
+
+  private static final String CONTENT_TYPE = "Content-Type";
+
+  public Response execute(String method, String url, Object body, Map<String, String> headers)
+      throws IOException {
+    HttpResponse httpResponse;
+    Request request;
+    switch (method) {
+      case "GET":
+        request = Request.get(url);
+        httpResponse = this.executeWithoutBody(request);
+        break;
+      case "POST":
+        request = Request.post(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "PUT":
+        request = Request.put(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "DELETE":
+        request = Request.delete(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "PATCH":
+        request = Request.patch(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "HEAD":
+        request = Request.head(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "OPTIONS":
+        request = Request.options(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      default:
+        throw new RuntimeException("Method is not supported");
+    }
+
+    ClassicHttpResponse classicHttpResponse = (ClassicHttpResponse) httpResponse;
+    int responseCode = httpResponse.getCode();
+
+    if (classicHttpResponse.getEntity() != null && classicHttpResponse.getEntity().getContent() != null) {
+      return new Response(responseCode,
+          IOUtils.toString(classicHttpResponse.getEntity().getContent(), StandardCharsets.UTF_8));
+    } else {
+      return new Response(responseCode, "");
+    }
+  }
+
+  private HttpResponse executeWithoutBody(Request request) throws IOException {
+    return request.execute().returnResponse();
+  }
+
+  private HttpResponse executeWithBody(Request request, Object body, Map<String, String> headers)
+      throws IOException {
+
+    // set body
+    String contentType = headers.get(CONTENT_TYPE);
+    if (ContentType.MULTIPART_FORM_DATA.getMimeType().equals(contentType)) {
+      Form form = Form.form();
+      Map<String, Object> bodyAsMap = (LinkedHashMap<String, Object>) body;
+      bodyAsMap.forEach((key, value) -> form.add(key, value.toString()));
+      request.bodyForm(form.build());
+    } else if (!contentType.isEmpty()) {
```

Comment:
perhaps we should document that the content-type is excepted if a body is set?

Reply:
Do we throw an exception here if the body is set without a content-type? 
We can also check that dependency in json-schema.
Or we just ignore the body in this case?

Reply:
I would perhaps use a default content type in this case (json?) and set the body
In Apache HTTP client for org.apache.hc.core5.http.io.entity.HttpEntities#create(java.lang.String) the default is text plain but I'm not sure it makes sense when calling APIs